src/Components/QuickSelect.tsx

```diff
@@ -1,202 +1,230 @@
+    );
+    if (recentlySelected.length > 6) {
+      recentlySelected.pop();
+    }
+    props.setRecentlySelected(recentlySelected);
+  }
+
+  function getDateSelectObj() {
+    return {
+      quickSelectContent: props.quickSelectContent,
+      handleClick,
+      dateRange: props.dateRange,
+      classes: props.classes,
+      termAnchorEl: props.termAnchorEl,
+      intervalAnchorEl: props.intervalAnchorEl,
+      setTermAnchorEl: props.setTermAnchorEl,
+      setIntervalAnchorEl: props.setIntervalAnchorEl,
+      quickSelectTerms: props.quickSelectTerms,
+      quickSelectIntervals: props.quickSelectIntervals,
+    };
+  }
+
+  return (
+    <Box className={props.boxClass}>
+      <Box className={props.classes.flexColumn}>
+        <Box ml={2} mt={2}>
+          <Typography variant="subtitle1" color="textPrimary">
+            Quick Select
+          </Typography>
+          <Box mt={1} className={props.classes.flexRow}>
+            <DateSelect {...getDateSelectObj()} />
+            <Box ml={1}>
+              <Button
+                onClick={() => apply()}
+                variant="contained"
+                color="primary"
+                className={props.classes.quickSelectApplyButton}
+              >
+                Apply
+              </Button>
+            </Box>
+          </Box>
+        </Box>
+        <Box ml={2} mt={2} className={props.classes.flexColumn}>
+          <Divider light />
+          <Box mt={1} />
+          <Typography color="textPrimary" variant="subtitle1">
+            Commonly used
+          </Typography>
+          <Box className={props.classes.flexRow}>
+            <Box className={props.classes.flexColumn}>
+              {props.commonlyUsedText.slice(0, 4).map((object) => (
+                <Box key={props.commonlyUsedText.slice(0, 4).indexOf(object)}>
+                  <Button
+                    onClick={() => handleClick(object)}
+                    color="primary"
+                    variant="text"
+                    size="small"
+                    disableFocusRipple={true}
+                    disableRipple={true}
+                    className={props.classes.quickSelectContainerButton}
+                  >
+                    {""}
+                  </Button>
+                  <Box mt={-3}>
+                    <Typography color="primary" variant="subtitle2">
+                      {props.dateRange
+                        .millisecondsToHumanized(
+                          DateRange.splitMilliseconds(object),
+                          TermContext.quickSelect
+                        )
+                        .join(" ")}
+                    </Typography>
+                  </Box>
+                </Box>
+              ))}
+            </Box>
+            <Box className={props.classes.flexColumn}>
+              {props.commonlyUsedText.slice(4, 8).map((object) => (
+                <Box key={props.commonlyUsedText.slice(4, 8).indexOf(object)}>
```

Comment:
That's an oddly specific slice code.  Can it be documented somehow or constants used so that future-you has an easier time understanding what's going on?

Reply:
not sure how i could make it a variable without moving the rendering into a function... Should I just document it instead?

Reply:
what do the 4 and 8 refer to?  Why 4 and 8?  Documentation would be a good start, of course.

Reply:
4 = half-length of quick select
8 = full length

I had to split them because they weren't formatting right